### PR TITLE
using templates instead of pointers, dispatch only once.

### PR DESCRIPTION
**Pull request by romainfrancois**
I've had a go at using templates and dispatch outside the loop. 

Not sure this is easier, simpler, etc ... I've tried to comment, happy to discuss farver 😆.

### CODE REVIEW

**Comment by thomasp85** [comment]
I think this makes sense, though it is not a setup I'd have thought of myself. Convoluted as it might be, it is easy enough to follow....

Have you done any timing comparisons?

**Comment by romainfrancois** [comment]
I have not. We need tools to easily benchmark different branches, etc ... ;-)

it's too much manual setup at the moment. `branchmark` ?

**Comment by thomasp85** [comment]
I think you should def make that 😀 I got a bit too much on my plate as it is with all that animation I need to discuss at useR

**Comment by romainfrancois** [comment]
I've done the lazy thing  and tweeted about it ... https://twitter.com/romain_francois/status/976096954554966016

**Comment by thomasp85** [comment]
I'm getting : 

```
Error in convert_c(as.matrix(colour), colourspace_match(from), colourspace_match(to)) : 
  Expecting a single string value: [type=integer; extent=1].
```

when I try to run the code in the readme benchmark - can you reproduce?

**Comment by romainfrancois** [comment]
Can't reproduce the problem. Actually I just realized that this code: 

```
test <- matrix(sample(256L, 30000, TRUE), ncol = 3) - 1L
timing <- microbenchmark(
  farver = convert_colour(test, 'rgb', 'lab'),
  grDevices = convertColor(test, 'sRGB', 'Lab', scale.in = 255)
)
```

pays for the conversion between the integer matrix that is passed in and the `NumericMatrix` that is declared in the C++ side.

File: DESCRIPTION [hunk 1]
```diff
@@ -11,7 +11,13 @@ Authors@R:
       person(given = "Berendea",
              family = "Nicolae",
              role = "aut",
-             comment = "Author of the ColorSpace C++ library"))
+             comment = "Author of the ColorSpace C++ library"),
+      person(given = "Romain", 
+             family = "François", 
+             role = "aut", 
+             email = "romain@purrple.cat",
+             comment = comment = c(ORCID = "0000-0002-2444-4226"))
```
Comment (thomasp85) [inline DESCRIPTION]: If you fix this little additional embarrassment I'll merge 😛

**Pull request merged**